sunpy/map/tests/test_mapbase.py

```diff
@@ -686,8 +686,17 @@ def test_resample_metadata(generic_map, sample_method, new_dimensions):
     Check that the resampled map has correctly adjusted metadata.
     """
     resampled_map = generic_map.resample(new_dimensions, method=sample_method)
-    assert resampled_map.meta['cdelt1'] == generic_map.meta['cdelt1']
-    assert resampled_map.meta['cdelt2'] == generic_map.meta['cdelt2']
+
+    scale_x = generic_map.data.shape[1] / resampled_map.data.shape[1]
```

Comment:
We seem to be missing an equivalent test for this for superpixel? In particular I can't see a test which checks superpixel's metadata when using a CD matrix?

Reply:
It's a good thing you asked for this, because it uncovered another bug: the `NAXIS1`/`NAXIS2` keywords weren't getting updated by `superpixel()`